@kiva/kv-shop/src/basket.ts

```diff
@@ -10,6 +13,32 @@ export function getBasketID() {
 	return getCookieValue('kvbskt');
 }
 
-export function setBasketID(basketId: string) {
+export function setBasketID() {
 	// TODO
 }
+
+export async function createBasket(apollo) {
+	try {
+		apollo.mutate({
+			mutation: gql`mutation createNewBasketForUser { shop { id createBasket } }`,
+		}).then(({ data }) => {
+			const newBasketId = data.shop?.createBasket ?? null;
+			if (newBasketId) {
+				document.cookie = `kvbskt=${newBasketId}; path=/; secure=true;`;
+			}
+		});
+	} catch (error) {
+		throw parseShopError(error);
+	}
+}
+
+export function hasBasketExpired(errorCode) {
+	return ['shop.invalidBasketId', 'shop.basketRequired'].includes(errorCode);
+}
+
+export async function handleInvalidBasketForDonation({ donationAmount, navigateToCheckout = false, apollo }) {
+	document.cookie = `kvbskt=; expires=${new Date(0).toUTCString()}; path=/;`;
```

Comment:
I think this would fail if used on the server due to the dependency on `document` and `window`.

We could either refactor to check for those before the code runs or maybe rename this to include "Client" in the name and add a doc block stating it's for clientside usage only.